### PR TITLE
implement __serialize and __tostring methods

### PR DESCRIPTION
**Pull request by olegrok**

### CODE REVIEW

**Comment by olegrok** [comment]
CI on 2.8 (I don't know why only it) is red and waits https://github.com/tarantool/icu-date/pull/25

File: test/test_add_spec.lua [hunk 1]
```diff
@@ -14,7 +14,7 @@ local function date_test(test, func)
     local format = icu_date.formats.iso8601()
 
     date:set_millis(1507836727123)
-    test:ok("2017-10-12T19:32:07.123Z" == date:format(format))
+    test:is("2017-10-12T19:32:07.123Z", date:format(format))
```
Comment (Totktonada) [inline test/test_add_spec.lua:17]: The `test:is()` function [accepts](https://github.com/tarantool/tarantool/blob/b0431cf8f47e9d081f6a402bc18edb1d6ad49847/src/lua/tap.lua#L142) the actual value as the first and the expected value as the second.

File: test/test_new_spec.lua [hunk 2]
```diff
@@ -3,16 +3,26 @@ require('strict').on()
...
               test:isnt(date, nil)
 end)
 
+test:test("serialize", function(test)
+              test:plan(3)
+              local date = icu_date.new()
+              test:isnt(date, nil)
+              date:set_millis(1234567891011)
+              test:is(tostring(date), '2009-02-13T23:31:31.011Z')
+              test:like(yaml.encode(date), '2009%-02%-13T23%:31%:31%.011Z')
```
Comment (Totktonada) [inline test/test_new_spec.lua:23]: Nit: Colon is not the [special character](https://www.lua.org/manual/5.3/manual.html#6.4.1).

**Comment by Totktonada** [comment]
Everything looks okay aside two minor comments above. Sure, we should push CI first, so waiting for #25, where it was implemented.

File: test/test_new_spec.lua (hunk 2 above)
Comment (olegrok) [reply test/test_new_spec.lua]: Fixed

File: test/test_add_spec.lua (hunk 1 above)
Comment (olegrok) [reply test/test_add_spec.lua]: Fixed

**Comment by Totktonada** [comment]
PR #25 is land to the master. I rebased this patchset.

**Review by Totktonada — APPROVED** [review]
LGTM.

**Pull request merged**